sdk/iot/core/src/az_iot_hub_client.c

```diff
@@ -107,3 +108,40 @@ AZ_NODISCARD az_result az_iot_hub_client_id_get(
 
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result
+az_iot_hub_client_properties_init(az_iot_hub_client_properties* properties, az_span buffer)
+{
+  AZ_PRECONDITION_NOT_NULL(properties);
+  AZ_PRECONDITION_VALID_SPAN(buffer, 0, false);
+
+  properties->_internal.properties = buffer;
+  properties->_internal.current_property = az_span_ptr(buffer);
```

Comment:
why do you do this as it's already in _internal.properties?

Reply:
setting the `current_property`?

Reply:
y

Reply:
I only see it used in a unit test, or am I blind?

Reply:
This could then get passed on to the parse functions and be used to parse through props. If the `properties` struct comes in and that isn't explicitly set, it could have some garbage value and the parse would attempt to iterate through garbage. If we explicitly set it, it can then be parsed with `az_iot_hub_client_properties_next()`. 

I believe the intended behavior also is that if someone is iterating through props and wants to reset, then call this function again and it acts as a reset.

Reply:
ok

Reply:
rename it iterator?

Reply:
I'd like to issue a new PR to name change to keep scope.

Reply:
Right: init initializes the enumeration.
We had the naming discussion already and we didn't want to call it "iterator".